### PR TITLE
Exception field

### PR DESCRIPTION
**Pull request by Avgor46**

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/// Extract exception info from stderr
+///
+/// # Arguments
+///
+/// * `stderr_list` - lines of stderr
+///
+/// # Return value
+///
+/// Exception info as a `ExecutionClass` struct
+pub fn exception_from_stderr(stderr_list: &[String]) -> Option<ExecutionClass> {
```
Comment (SweetVishnya) [inline src/util.rs:13]: maybe we should name file cpp.rs and rename function to specify that it is a cpp exception

File: src/util.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        .position(|line| rexception.is_match(line))
+    {
+        let instance = rexception
+            .captures(&stderr_list[pos])
+            .unwrap()
+            .get(1)
+            .unwrap()
+            .as_str();
+        let message = if let Some(element) = stderr_list.get(pos + 1) {
+            let rwhat = Regex::new(r"  what\(\):  (.+)").unwrap();
```
Comment (SweetVishnya) [inline src/util.rs:26]: `r"what\(\): +(.+)"`

File: src/util.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            if let Some(cap) = rwhat.captures(element) {
+                cap.get(1).unwrap().as_str()
```
Comment (SweetVishnya) [inline src/util.rs:28]: trim

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: tests/tests.rs [hunk 4]
```diff
@@ -2681,6 +2774,7 @@ fn test_casr_san_sigbus() {
 }
 
 #[test]
+#[ignore]
```
[the remark this answers is not in the document]
Comment (Avgor46) [reply tests/tests.rs]: removed

File: src/util.rs (hunk 1 above)
Comment (Avgor46) [reply src/util.rs]: exception_from_stderr -> cpp_exception_from_stderr

File: src/util.rs (hunk 2 above)
Comment (Avgor46) [reply src/util.rs]: done

File: src/util.rs (hunk 1 above)
Comment (Avgor46) [reply src/util.rs]: util.rs will be used for rust panic too

**Review by SweetVishnya — APPROVED** [review]

**Pull request merged**